src/KDSoapServer/KDSoapServerSocket.cpp

```diff
@@ -414,22 +424,126 @@ void KDSoapServerSocket::handleRequest(const QMap<QByteArray, QByteArray> &httpH
+    if (!rangeHeader.startsWith("bytes=")) {
+        // Section 3.1 (RFC 7233): Ignore a range header field that contains a range unit the server does not understand
+        return requestedRanges;
+    }
+
+    const QByteArray rangeSpec = rangeHeader.mid(6);
+    if (!std::all_of(rangeSpec.begin(), rangeSpec.end(), [](char c) {
+            return std::isdigit(c) || c == '-' || c == ',' || c == ' ' || c == '\t';
+        })) {
+        // Section 14.3.2: A server that receives a malformed Range header field
+        // MAY ignore it and serve the whole resource, or respond with a 416
+        // We ignore it and send a 200.
+        return requestedRanges;
+    }
+
+    const QByteArrayList rangeSpecs = rangeHeader.mid(6).split(',');
+    for (const QByteArray &spec : rangeSpecs) {
+        const QList<QByteArray> parts = spec.trimmed().split('-');
+        if (parts.size() != 2) {
+            requestedRanges.clear();
+            break; // Malformed — Status code 416
+        }
+
+        bool okStart = true, okEnd = true;
+        const QByteArray &startStr = parts[0];
+        const QByteArray &endStr = parts[1];
+        int start = -1, end = -1;
+
+        if (!startStr.isEmpty())
+            start = startStr.toInt(&okStart);
+        if (!endStr.isEmpty())
+            end = endStr.toInt(&okEnd);
+
+        if ((!okStart && !startStr.isEmpty()) || (!okEnd && !endStr.isEmpty()) || (start < 0 && end < 0)) {
+            requestedRanges.clear();
+            break; // Status code 416
+        }
+
+        // Handle suffix-byte-range-spec (e.g. bytes=-500)
+        if (start < 0 && end >= 0) {
+            start = qMax(0, fileSize - end);
+            end = fileSize - 1;
+        }
+
+        // Handle open-ended range (e.g. bytes=100-)
+        else if (start >= 0 && end < 0) {
+            end = fileSize - 1;
+        }
+
+        // Disallow invalid ranges (start > end)
+        if (start > end || start >= fileSize) {
+            continue; // Skip unsatisfiable range, per RFC
+        }
+
+        // Truncate end to EOF if necessary
+        end = qMin(end, fileSize - 1);
+        requestedRanges.append(qMakePair(start, end));
+    }
+
+    // Sort and coalesce ranges
+    if (requestedRanges.size() > 1) {
+        std::sort(requestedRanges.begin(), requestedRanges.end());
+
+        for (int i = 0; i < requestedRanges.size() - 1;) {
+            auto &current = requestedRanges[i];
+            auto &next = requestedRanges[i + 1];
+            if (next.first <= current.second + 1) {
+                current.second = qMax(current.second, next.second);
+                requestedRanges.remove(i + 1);
+            } else {
+                ++i;
+            }
+        }
+    }
+
+    if (requestedRanges.isEmpty() || rangeSpecs.size() > 5) {
+        // No ranges found in header or
+        // Section 6.1 (RFC 7233): Too many ranges — reject
+        write("HTTP/1.1 416 Range Not Satisfiable\r\nContent-Length: 0\r\n\r\n");
```

Comment:
shouldn't the caller take care of this write()? I would expect this method to only do the parsing (and to be `const`).